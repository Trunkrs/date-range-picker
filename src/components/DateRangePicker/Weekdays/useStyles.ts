import createStyles from '@material-ui/core/styles/createStyles'
import makeStyles from '@material-ui/core/styles/makeStyles'
import { Theme } from '@material-ui/core/styles'

const styles = ({ spacing, palette }: Theme) =>
  createStyles({
    weekdays: {
      display: 'grid',
      gridTemplateColumns: `0px repeat(5, ${spacing(5)}px) 0px`,
    },
    weekdayName: {
      color: palette.trunkrs.grayscale.darker,
      position: 'relative',
      textAlign: 'center',
      width: '100%',
      height: spacing(5),
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      overflow: 'hidden',
    },
    showWeekends: {
      gridTemplateColumns: `repeat(7, ${spacing(5)}px) !important`,
    },
  })

export default makeStyles(styles)
