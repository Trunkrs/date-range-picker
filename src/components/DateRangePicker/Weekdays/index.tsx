import React, { useMemo } from 'react'

import clsx from 'clsx'
import addDays from 'date-fns/addDays'
import startOfWeek from 'date-fns/startOfWeek'
import format from 'date-fns/format'

import Typography from '../../Typography'

import useStyles from './useStyles'

export interface WeekdaysProps {
  showWeekends?: boolean
}

export const Weekdays: React.FC<WeekdaysProps> = ({ showWeekends = true }) => {
  const classes = useStyles()

  const firstDOW = startOfWeek(new Date())
  const shortWeekDaysArray = useMemo(
    () =>
      Array.from(Array(7)).map((_e, i) =>
        format(addDays(firstDOW, i), 'EEEEEE'),
      ),
    [firstDOW],
  )

  return (
    <div
      className={clsx(classes.weekdays, showWeekends && classes.showWeekends)}
    >
      {shortWeekDaysArray.map((weekdayName) => (
        <Typography
          variant="textBold"
          className={classes.weekdayName}
          key={weekdayName}
        >
          {weekdayName}
        </Typography>
      ))}
    </div>
  )
}

export default React.memo(Weekdays)
