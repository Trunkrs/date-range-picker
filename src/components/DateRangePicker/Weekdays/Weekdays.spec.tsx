import React from 'react'

import { createShallow } from '@material-ui/core/test-utils'

import Weekdays, { WeekdaysProps } from '.'

const createProps = (merge: Partial<WeekdaysProps> = {}): WeekdaysProps =>
  ({
    ...merge,
  } as WeekdaysProps)

describe('<Weekdays />', () => {
  const shallow = createShallow()

  it('should match snapshot', () => {
    const wrapper = shallow(<Weekdays {...createProps()} />)

    expect(wrapper).toMatchSnapshot()
  })
})
