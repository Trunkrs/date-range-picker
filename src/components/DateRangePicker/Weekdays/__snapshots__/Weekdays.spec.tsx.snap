// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weekdays /> should match snapshot 1`] = `
<div
  className="makeStyles-weekdays-1 makeStyles-showWeekends-3"
>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="Su"
    variant="textBold"
  >
    Su
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="Mo"
    variant="textBold"
  >
    Mo
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="Tu"
    variant="textBold"
  >
    Tu
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="We"
    variant="textBold"
  >
    We
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="Th"
    variant="textBold"
  >
    Th
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="Fr"
    variant="textBold"
  >
    Fr
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="makeStyles-weekdayName-2"
    key="Sa"
    variant="textBold"
  >
    Sa
  </WithStyles(Typography)>
</div>
`;
